skrub/_reporting/_table_report.py

```diff
@@ -134,31 +140,38 @@ def __init__(
         self.dataframe = dataframe
         self.verbose = verbose
         self.max_plot_columns = max_plot_columns
+        self.max_association_columns = max_association_columns
+        self.n_columns = sbd.shape(self.dataframe)[1]
+
+        if self.max_plot_columns is None:
+            self.max_plot_columns = self.n_columns
+        if self.max_association_columns is None:
+            self.max_association_columns = self.n_columns
 
     def __repr__(self):
         return f"<{self.__class__.__name__}: use .open() to display>"
 
     @functools.cached_property
-    def _summary_with_plots(self):
-        return summarize_dataframe(
-            self.dataframe, with_plots=True, title=self.title, **self._summary_kwargs
-        )
+    def _lightify_summary(self):
+        if self.max_plot_columns >= self.n_columns:
+            with_plots = True
+        else:
+            with_plots = False
+        if self.max_association_columns >= self.n_columns:
+            with_associations = True
+        else:
+            with_associations = False
 
-    @functools.cached_property
-    def _summary_without_plots(self):
         return summarize_dataframe(
-            self.dataframe, with_plots=False, title=self.title, **self._summary_kwargs
+            self.dataframe,
+            with_plots=with_plots,
+            with_associations=with_associations,
```

Comment:
We could simplify the lines above like:
```suggestion
            with_plots=self.max_plot_columns >= self.n_columns,
            with_associations=self.max_association_columns >= self.n_columns,
```